src/ImageProcessor.Web.Plugins.AzureBlobCache/AzureBlobCache.cs

```diff
@@ -519,6 +519,42 @@ public override void RewritePath(HttpContext context)
         }
 
         /// <summary>
+        /// Tries to set IfModifiedSince howver this crashes when context.Request.Headers["If-Modified-Since"] exists, 
+        /// but is not parsable.It's not parsable when it comes from Google Bot as UTC eg Sun, 27 Nov 2016 20:01:45 UTC
+        /// so DateTime.TryParse. If it returns false, then log the error.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="request"></param>
+        private static void TrySetIfModifiedSinceDate(HttpContext context, HttpWebRequest request)
+        {
+            DateTime ifModifiedDate;
+
+            string ifModifiedFromRequest = context.Request.Headers["If-Modified-Since"];
+
+            if (DateTime.TryParse(ifModifiedFromRequest, out ifModifiedDate))
+            {
+                request.IfModifiedSince = ifModifiedDate;
+            }
+            else
+            {
+                if (ifModifiedFromRequest.ToLower().Contains("utc"))
+                {
+                    ifModifiedFromRequest = ifModifiedFromRequest.ToLower().Replace("utc", string.Empty);
+
+                    if (DateTime.TryParse(ifModifiedFromRequest, out ifModifiedDate))
+                    {
+                        request.IfModifiedSince = ifModifiedDate;
+                    }
+                }
+                else
+                {
+                    ImageProcessorBootstrapper.Instance.Logger.Log<string>("Unable to parse date " +
```

Comment:
The type param here should be the parent class. Also might be an idea to add information about the request url